hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -854,6 +865,43 @@ private String reconfSlowDiskParameters(String property, String newVal)
     }
   }
 
+  private String reconfDfsUsageParameters(String property, String newVal)
+      throws ReconfigurationException {
+    String result = null;
+    try {
+      LOG.info("Reconfiguring {} to {}", property, newVal);
+      if (property.equals(FS_DU_INTERVAL_KEY)) {
+        Preconditions.checkNotNull(data, "FsDatasetSpi has not been initialized.");
+        long interval = (newVal == null ? FS_DU_INTERVAL_DEFAULT :
+            Long.parseLong(newVal));
+        result = Long.toString(interval);
+        List<FsVolumeImpl> volumeList = data.getVolumeList();
+        for (FsVolumeImpl fsVolume : volumeList) {
+          Map<String, BlockPoolSlice> blockPoolSlices = fsVolume.getBlockPoolSlices();
+          for (Entry<String, BlockPoolSlice> entry : blockPoolSlices.entrySet()) {
+            entry.getValue().updateDfsUsageConfig(interval, null);
+          }
```

Comment:
We can make it simpler.
```suggestion
          for (BlockPoolSlice bp : blockPoolSlices.values()) {
            bp.updateDfsUsageConfig(interval, null);
          }
```